python/test/nex2ecl.py

```diff
@@ -0,0 +1,104 @@
+import datetime
+import unittest
+import os
+import tempfile
+import shutil
+import nex
+import ecl.ecl as ecl
+
+
+class TestLoad(unittest.TestCase):
+    def setUp(self):
+        self.plt = nex.load('test-data/SPE1.plt')
+        self.plt_delayed = nex.load('test-data/SPE1_delayedWell.plt')
+        self.work_area_path = tempfile.mkdtemp()
+        self.cwd = os.getcwd()
+        os.chdir(self.work_area_path)
+
+    def tearDown(self):
+        os.chdir(self.cwd)
+        shutil.rmtree(self.work_area_path)
+
+    def test_ecl_sum(self):
+        ecl_sum = nex._nex2ecl(self.plt, 'ECL_CASE', format=False,
+                               field_name='FIELD')
+        self.assertTrue(isinstance(ecl_sum, ecl.ecl_sum.EclSum))
+
+        start_date = ecl_sum.start_date
+        self.assertEqual(start_date, datetime.date(1980, 1, 1))
+        self.assertEqual(list(ecl_sum.wells()), ['1', '2'])
+        self.assertEqual(ecl_sum.unit('FOPR'), 'SM3/DAY')
+        self.assertEqual(len(ecl_sum.keys()), 53)
+        self.assertEqual(len(ecl_sum), 10)
+
```

Comment:
Although the classes are certinaly the most important to test I also find it valuable to have somebasic testing of the binary script:
```python
#Untested!
def test_script(self):
     script = os.path.abspath( os.path.join( os.path.dir( __file__), "../applications/nex2ecl"))
     with TestAreaContext("test_ares") as t:
             t.copy_file("'test-data/SPE1.plt")
             subprocess.check_call([script, ....])
             ecl_case = EclSum( "SPE1")
```

Reply:
This is problematic because the application would have to be installed into the build directory. This could work if the building was rewritten to use setuptools as the main driver for the application installation. At the moment CMake is responsible for installing it. Since we might switch to use argparse rather than getopt for the application, the testing could be implemented in that PR.